lib/charms/tempo_k8s/v1/tracing.py

```diff
@@ -547,37 +537,65 @@ def get_all_endpoints(
         """Ingester endpoint for the ``otlp_grpc`` protocol."""
-        return self._get_ingester(relation or self._relation, protocol="otlp_grpc")
+        return self._get_ingester(relation or self._relation, protocol="otlp_grpc", ssl=ssl)
+
+    def otlp_http_endpoint(
+        self, relation: Optional[Relation] = None, ssl: bool = False
+    ) -> Optional[str]:
+        """Ingester endpoint for the ``otlp_http`` protocol.
 
-    def otlp_http_endpoint(self, relation: Optional[Relation] = None) -> Optional[str]:
-        """Ingester endpoint for the ``otlp_http`` protocol."""
-        return self._get_ingester(relation or self._relation, protocol="otlp_http")
+        The provided endpoint does not contain the endpoint suffix. If the instrumenting library needs the full path,
+        your endpoint code needs to add the ``/v1/traces`` suffix.
+        """
+        return self._get_ingester(relation or self._relation, protocol="otlp_http", ssl=ssl)
 
-    def zipkin_endpoint(self, relation: Optional[Relation] = None) -> Optional[str]:
-        """Ingester endpoint for the ``zipkin`` protocol."""
-        return self._get_ingester(relation or self._relation, protocol="zipkin")
+    def zipkin_endpoint(
+        self, relation: Optional[Relation] = None, ssl: bool = False
+    ) -> Optional[str]:
+        """Ingester endpoint for the ``zipkin`` protocol.
 
-    def tempo_endpoint(self, relation: Optional[Relation] = None) -> Optional[str]:
+        The provided endpoint does not contain the endpoint suffix. If the instrumenting library needs the full path,
+        your endpoint code needs to add the ``/api/v2/spans`` suffix.
+        """
+        return self._get_ingester(relation or self._relation, protocol="zipkin", ssl=ssl)
+
+    def tempo_endpoint(
+        self, relation: Optional[Relation] = None, ssl: bool = False
+    ) -> Optional[str]:
         """Ingester endpoint for the ``tempo`` protocol."""
-        return self._get_ingester(relation or self._relation, protocol="tempo")
+        return self._get_ingester(relation or self._relation, protocol="tempo", ssl=ssl)
+
+    def jaeger_http_thrift_endpoint(
+        self, relation: Optional[Relation] = None, ssl: bool = False
+    ) -> Optional[str]:
+        """Ingester endpoint for the ``jaeger_http_thrift`` protocol.
 
-    def jaeger_http_thrift_endpoint(self, relation: Optional[Relation] = None) -> Optional[str]:
-        """Ingester endpoint for the ``jaeger_http_thrift`` protocol."""
-        return self._get_ingester(relation or self._relation, "jaeger_http_thrift")
+        The provided endpoint does not contain the endpoint suffix. If the instrumenting library needs the full path,
+        your endpoint code needs to add the ``/api/traces`` suffix.
+        """
+        return self._get_ingester(relation or self._relation, "jaeger_http_thrift", ssl=ssl)
 
-    def jaeger_grpc_endpoint(self, relation: Optional[Relation] = None) -> Optional[str]:
+    def jaeger_grpc_endpoint(
```

Comment:
does ssl make sense with grpc? we shouldn't add it as an arg to this method if we don't plan on doing anything with it

Reply:
(same with the other grpc protocol)

Reply:
On second thought the path GRPC dependencies use is the same regardless if SSL is used or not so I removed the parameter.